lib/grammars.coffee

```diff
@@ -137,7 +137,11 @@ module.exports =
   Go:
     "File Based":
       command: "go"
-      args: (context) -> ['run', context.filepath]
+        args: (context) ->
+          if context.filepath.match(/_test.go/)
+            ['test', '']
+          else ['run', context.filepath]
+      workingDirectory: atom.workspace.getActivePaneItem()?.buffer?.file?.getParent?().getPath?()
```

Comment:
I'm a bit unsure what you're trying to do here. Is this reaching the root of the project for the working Directory?